ext/date/php_date.c

```diff
@@ -1065,15 +1065,16 @@ PHP_FUNCTION(strtotime)
 PHPAPI void php_mktime(INTERNAL_FUNCTION_PARAMETERS, int gmt)
 {
 	zend_long hou, min, sec, mon, day, yea;
-	zend_bool min_is_null = 1, sec_is_null = 1, mon_is_null = 1, day_is_null = 1, yea_is_null = 1;
+	zend_bool hou_is_null = 1, min_is_null = 1, sec_is_null = 1, mon_is_null = 1, day_is_null = 1, yea_is_null = 1;
 	timelib_time *now;
 	timelib_tzinfo *tzi = NULL;
 	zend_long ts, adjust_seconds = 0;
 	int epoch_does_not_fit_in_zend_long;
 
+	/* at least one arg is required, although it does not matter which one is given */
```

Comment:
If we do this, I'd prefer to just accept no arguments as well. Doesn't really hurt anyone, does it?

Reply:
No, it wouldn't, but @derickr still [may not like it](https://externals.io/message/26152).

Reply:
Okay ... we can enforce at least one arg, but this isn't the way to do it. I'm surprised you didn't get any assertion failures. This will also still allow something like `mktime(null)`.

If we want to enforce at least one argument, I'd expect it to look something like this:

```
if (hou_is_null && min_is_null && ...) {
    zend_argument_count_error(...);
    RETURN_THROWS();
}
```

Reply:
I have updated the implementation now. However, it feels a bit silly to disallow calling these functions without parameter, because the same could be accomplished with (gm)time() more efficiently.